youtube_dl/extractor/morningstar.py

```diff
@@ -16,32 +14,27 @@ class MorningstarIE(InfoExtractor):
             'id': '615869',
             'ext': 'mp4',
             'title': 'Get Ahead of the Curve on 2013 Taxes',
-            'description': "Vanguard's Joel Dickson on managing higher tax rates for high-income earners and fund capital-gain distributions in 2013.",
+            'description': 'md5:0f71dedfd54bb3dca09e6474e2b624f7',
             'thumbnail': r're:^https?://.*m(?:orning)?star\.com/.+thumb\.jpg$'
         }
     }
 
     def _real_extract(self, url):
-        mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
+        video_id = self._match_id(url)
 
         webpage = self._download_webpage(url, video_id)
-        title = self._html_search_regex(
-            r'<h1 id="titleLink">(.*?)</h1>', webpage, 'title')
+        title = get_element_by_id('titleLink', webpage)
```

Comment:
Should we change `get_element_by_id` to handle `fatal=True` cases? Or should I just raise an `ExtractorError` whenever title extraction fails?